src/builder_state.rs

```diff
@@ -579,11 +579,22 @@ impl<TYPES: NodeType> BuilderProgress<TYPES> for BuilderState<TYPES> {
             .for_each(|txn| {
                 if let Entry::Occupied(txn_info) = self.tx_hash_to_available_txns.entry(*txn) {
                     self.timestamp_to_tx.remove(&txn_info.get().0);
-                    self.included_txns.insert(*txn);
+                    self.included_txns.insert(*txn_info.key(), txn_info.get().0);
                     txn_info.remove_entry();
                 }
             });
 
+        // clear the txns from included txns which are older than current timestamp by 3 secs
```

Comment:
What? No. I meant, we could clear after... days? Hours, maybe, at the smallest increment.

Reply:
We can do 2 hours. But... maybe let's try this instead: https://espresso.zulipchat.com/#narrow/stream/419378-Block-Builder/topic/Staging.20performance.20issues/near/440153803 with the <unit of time> configurable, defaulting to 1 hour.